Tasks/NuGetInstaller/VsoNuGetHelper.ps1

```diff
@@ -0,0 +1,157 @@
+}
+
+function SetNewCredentialElement
+{
+    param(
+        [xml] $nugetConfig,
+        [System.Xml.XmlElement] $credentialsSection,
+        [string] $credentialKey,
+        [string] $password
+    )
+
+    $newCredential = $nugetConfig.CreateElement($credentialKey)
+
+    $userNameElement= $nugetConfig.CreateElement("add")
+    [void]$userNameElement.SetAttribute("key", "UserName")
+    [void]$userNameElement.SetAttribute("value", "VssSessionToken")
+    [void]$newCredential.AppendChild($userNameElement)
+
+    $passwordElement = $nugetConfig.CreateElement("add")
+    [void]$passwordElement.SetAttribute("key", "Password")
+    $encryptedPassword = EncryptNuGetPassword $password
+    [void]$passwordElement.SetAttribute("value", $encryptedPassword)
+    [void]$newCredential.AppendChild($passwordElement)
+
+    [void]$credentialsSection.AppendChild($newCredential)
+}
+
+function ShouldGetCredentialsForFeed
+{
+    param(
+        [Uri]$feedUri
+    )
+
+    if(-not ($feedUri.Scheme.Equals("http","InvariantCultureIgnoreCase") -or  $feedUri.Scheme.Equals("https","InvariantCultureIgnoreCase")))
+    {
+        return $false
+    }
+
+    $WebRequestObject = [System.Net.HttpWebRequest] [System.Net.WebRequest]::Create($feedUri);
+    try
+    {
+        $ResponseObject = [System.Net.HttpWebResponse] $WebRequestObject.GetResponse();
+    }
+    catch [Net.WebException] 
+    {
+        $ResponseObject = [System.Net.HttpWebResponse] $_.Exception.Response
+    }
+    
+    if($ResponseObject.StatusCode -eq 401)
+    {
+        $tfsHeaderPresent = $false
+        $vssHeaderPresent = $false
+        foreach($key in $ResponseObject.Headers.AllKeys)
+        {
+            if($tfsHeaderPresent -or $key.Contains("X-TFS"))
+            {
+                $tfsHeaderPresent = $true    
+            }
+
+            if($key.Contains("X-VSS"))
+            {
+                $vssHeaderPresent = $true
+            }
+
+            if($tfsHeaderPresent -and $vssHeaderPresent)
+            {
+                break
+            }
+        }
+
+        [void]$ResponseObject.Close();
+        return $tfsHeaderPresent -and $vssHeaderPresent
+    }
+    
+    return $false
+}
+
+function SetCredentialsNuGetConfigAndSaveTemp
+{
+    param(
+        [xml] $nugetConfig,
+        [string] $accessToken,
+        [uri]$targetFeed
+    )
+
+    $credentialsSection = $nugetConfig.SelectSingleNode("configuration/packageSourceCredentials")
+    if($credentialsSection -eq $null)
+    {
+        #add the packageSourceCredentials node
+        $credentialsSection = $nugetConfig.CreateElement("packageSourceCredentials")
+        [void]$nugetConfig.configuration.AppendChild($credentialsSection)
+    }
+    
+    $nugetSources = $nugetConfig.SelectNodes("configuration/packageSources/add")
+    foreach($nugetSource in $nugetSources)
+    {
+        if($targetFeed -and -not $targetFeed.ToString().Equals($nugetSource.value,"InvariantCultureIgnoreCase"))
```

Comment:
Consider adding parentheses here for readability. E.g. 
if ($targetFeed -and (-not [...]))